src/hoc/withDepartures.js

```diff
@@ -0,0 +1,88 @@
+import {inject, observer} from "mobx-react";
+import {app} from "mobx-app";
+import React from "react";
+import withRoute from "./withRoute";
+import gql from "graphql-tag";
+import {Query} from "react-apollo";
+import {get} from "lodash";
+import getDay from "date-fns/get_day";
+import parse from "date-fns/parse";
+import isWithinRange from "date-fns/is_within_range";
+
+const departuresQuery = gql`
+  query routeDepartures(
+    $routeId: String
+    $direction: String
+    $dayType: String
+    $stopId: String
+  ) {
+    allDepartures(
```

Comment:
I would add here       orderBy: DEPARTURE_ID_ASC 
because then you don't have to sort the result